Analysis/src/EventTags.cpp

```diff
@@ -48,49 +59,45 @@ EventTags EventTagCreator::CreateEventTags(const std::vector<DataId>& dataIds_ba
         { DiscriminatorWP::Tight, num_btag_tight}
     };
 
-    boost::optional<DiscriminatorWP> vbf_tag;
-    if(vbf_tag_raw > 0) vbf_tag = static_cast<DiscriminatorWP>(vbf_tag_raw);
+    std::map<DiscriminatorWP, float > btag_weights = {};
+    if(btag_weight_Loose.size()!=0 || btag_weight_Medium.size()!=0 || btag_weight_Tight.size()!=0) btag_weights = {
+      { DiscriminatorWP::Loose, btag_weight_Loose.at(0) },
```

Comment:
- define map as `std::map<DiscriminatorWP, const std::vector<float>* >`;
- create a lambda to get weight given unc and vector as arguments (like this it will be universal for all weights and unc variations).

Reply:
Which type is used to pass the unc?

Reply:
`UncertaintyScale` which is defined [here](https://github.com/hh-italian-group/h-tautau/blob/master/Core/include/AnalysisTypes.h#L38)

The definition of how up/down weights are stored in anaTuple is defined [here](https://github.com/hh-italian-group/hh-bbtautau/blob/master/Analysis/src/AnaTuple.cpp#L96-L111). The new lambda should basically perform an inverse operation and return the weight for the given scale.